test/e2e/externally_provisioned_test.go

```diff
@@ -79,14 +80,28 @@ var _ = Describe("Create as externally provisioned, deprovision", Label("require
 			}, &bmh)
 			Expect(err).NotTo(HaveOccurred())
 
-			By("checking that the BMH was not inspected or deployed")
+			By("Checking that the BMH was not inspected or deployed")
 			Expect(bmh.Status.OperationHistory.Inspect.Start.IsZero()).To(BeTrue())
 			Expect(bmh.Status.OperationHistory.Provision.Start.IsZero()).To(BeTrue())
 
+			By("Removing the ExternallyProvisioned field")
+			bmh.Spec.ExternallyProvisioned = false
+
+			err = clusterProxy.GetClient().Update(ctx, &bmh)
+			Expect(err).NotTo(HaveOccurred())
+
+			Eventually(func(g Gomega) {
+				err = clusterProxy.GetClient().Get(ctx, kclient.ObjectKeyFromObject(&bmh), &bmh)
+				g.Expect(bmh.Status.Provisioning.State).To(Equal(metal3api.StateProvisioned))
```

Comment:
The way the test is written, the expected state is not Provisioned. Or even: it make become Provisioned briefly, then BMO detects that there is no `image` and no `customDeploy` and deprovisions.